scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -1,48 +1,139 @@
 package scalafix
 package internal.config
 
-import metaconfig.ConfDecoder
+import metaconfig._
 import org.langmeta.Symbol
+
 import scalafix.internal.util.SymbolOps
-import metaconfig.annotation.Description
-import metaconfig.annotation.ExampleValue
-import metaconfig.generic
+import metaconfig.annotation.{DeprecatedName, Description, ExampleValue}
 import metaconfig.generic.Surface
 
+case class DisablePart(
+    @Description("The symbol that indicates a 'safe' block. Optional value.")
+    @DeprecatedName(
+      "unless",
+      "Renamed to unlessInsideBlock",
+      "0.6.0"
+    )
+    unlessInsideBlock: Option[Symbol.Global],
+    @Description(
+      "When true also blocks symbols in the generated code. Default to false. " +
+        "Ignored if unlessInsideBlock is specified.")
+    unlessSynthetic: Boolean,
+    @Description("The unsafe that are banned unless inside a 'safe' block")
+    symbols: List[CustomMessage[Symbol.Global]])
+
+object DisablePart {
+  implicit val surface: Surface[DisablePart] =
+    generic.deriveSurface[DisablePart]
+  // NOTE(olafur): metaconfig.generic.deriveDecoder requires a default base values.
+  // Here we require all fields to be provided by the user so we write the decoder manually.
+  implicit val reader: ConfDecoder[DisablePart] =
+    ConfDecoder.instanceF[DisablePart] {
+      case c: Conf.Obj =>
+        (c.getOption[Symbol.Global]("unlessInsideBlock", "unless") |@|
+          c.getOrElse[Boolean]("unlessSynthetic")(false) |@|
+          c.get[List[CustomMessage[Symbol.Global]]]("symbols")).map {
+          case ((a @ Some(_), _), c) => DisablePart(a, false, c)
+          // when unlessInsideBlock is specified we ignore unlessSynthetic
+          case ((None, b), c) => DisablePart(None, b, c)
+        }
+      case _ => Configured.NotOk(ConfError.message("Wrong config format"))
+    }
+}
+
 case class DisableConfig(
-    @Description("The list of symbols to disable.")
-    @ExampleValue(
-      """[
-        |  # With custom message (recommended)
-        |  {
-        |    symbol = "scala.Predef.any2stringadd"
-        |    message = "Use explicit toString before calling +"
-        |  }
-        |  {
-        |    symbol = "scala.Any"
-        |    message = "Explicitly type annotate Any if this is intentional"
-        |  }
-        |  # Without custom message (discouraged)
-        |  "com.Lib.implicitConversion"
-        |]""".stripMargin)
-    symbols: List[CustomMessage[Symbol.Global]] = Nil
+    @Description("The list of disable configurations")
+    @ExampleValue("""[
+                    |  { // default mode
```

Comment:
We should preserve the old syntax for `Disable.symbols`, the new layout is not user-friendly
```
Disable.parts = [
  { symbols = [
    ...
  ]}
]
```

Instead of `parts`, I would prefer
```
Disable.symbols = [...]
Disable.unlessSynthetic = [ ... ]
Disable.unlessInsideBlock = [ ... ]
```

Reply:
I'm confused by how `unlessSynthetic` works here. Is it a "disable this even if it is synthetic" setting? Neither `symbols` nor `unless` should operate on synthetic symbols.